rfcs/high-contrast-theme.md

```diff
@@ -0,0 +1,125 @@
+# RFC: High Contrast theme
+
+---
+
+_List contributors to the proposal here_: @miroslavstastny
+
+## Summary
+
+High contrast theme should use system colors if user enables forced colors in their Windows settings.
+
+## Problem statement
+
+<!--
+Why are we making this change? What problem are we solving? What do we expect to gain from this?
+
+This section is important as the motivation or problem statement is indepenent from the proposed change. Even if this RFC is not accepted this Motivation can be used for alternative solutions.
+
+In the end, please make sure to present a neutral Problem statement, rather than one that motivates a particular solution
+-->
+
+`react-components` support High Contrast theme. There are currently two problems with the theme:
+
+1. Even if user configures their Windows to use high contrast, they need to explicitly set High Contrast (HC) theme in Fluent UI (FUI) application.
+2. Fluent UI High Contrast theme uses colors chosen by designers. Those are always light on dark with hardcoded colors - users cannot choose to use dark on light HC theme or customise the colors used - this might limit users with specific visual impairments.
+
+This RFC does not address the first issue - after the change, users will still need to opt-in to use HC theme in FUI application. This can be discussed and addressed in a separate RFC.
+
+The main topic of the RFC is the second issue - HC FUI theme driven by System HC color theme.
+
+## Detailed Design or Proposal
```

Comment:
Just I am trying to understand and catch up the whole problematic with HC mode. 

I fully agree with the proposal to use system high contrast mode if it is enabled. 

But I have more questions if system high contrast mode cannot be used or detected.

When I tried now Teams in Electron with "High Contact White" set in Windows, then Electron Teams app is still black. High probably it is issue of Electron. 
When this happen should we then tell user about this state? show some alert/info banner about not ability to use system HC? (otherwise user doesn't know what is happening) 

If we cannot use system HC should we then provide equivalent for both white and black theme? and should have these custom themes have the same default color as system ones?

Reply:
HC will use system colors and in cases where these colors are not available, it will fall back to the existing white-on-black HC colors which we have now.